src/components/AppConfig/Vector.tsx

```diff
@@ -1,29 +1,31 @@
 import React from "react";
 
-import { Checkbox, Field, FieldSet, Input, Select, useStyles2 } from "@grafana/ui";
+import { Checkbox, Field, FieldSet, InlineField, InlineSwitch, Input, LegacyForms, Select } from "@grafana/ui";
 
 import { testIds } from "components/testIds";
-import { getStyles } from "./AppConfig";
+import { AppPluginSettings, Secrets, SecretsSet } from "./AppConfig";
+
+import { BasicAuthConfig } from "./AuthSettings/BasicAuth";
+import { TokenAuthConfig } from "./AuthSettings/TokenAuth";
 
 export interface VectorSettings {
   // Whether the vector service should be enabled.
   enabled?: boolean;
   // The model used by the embedder and for embeddings stored in the store.
   model?: string;
   // Embedder settings.
-  embed?: EmbedderSettings;
+  embedType?: string;
```

Comment:
Hmm I'd be tempted to keep this as the `EmbedderSettings` object with just the `type` key, in case we want to add more to it in future (e.g. supporting more than OpenAI and storing some data for the new embedder type).